### PR TITLE
Use JSON instead of Pickle

### PR DESCRIPTION
**Pull request by bethbrains**
- All new keychain details are serialized using json, not pickle.

### CODE REVIEW

**Comment by prescod** [comment]
@bethbrains 

We're going to have to support Dates and DateTimes.

https://github.com/SFDO-Tooling/CumulusCI/actions/runs/3230706675/jobs/5289461956#step:8:41

We need to use the `default=` parameter:

https://docs.python.org/3/library/json.html#json.dump

We should also do some kind of survey to figure out what other types we might need to support.

File: cumulusci/core/keychain/encrypted_file_project_keychain.py [hunk 1]
```diff
@@ -43,17 +45,18 @@
...
 scratch_org_class = os.environ.get("CUMULUSCI_SCRATCH_ORG_CLASS")
 if scratch_org_class:
     scratch_org_factory = import_global(scratch_org_class)  # pragma: no cover
 else:
     scratch_org_factory = ScratchOrgConfig
 
 
+def simplify(x):
+    if isinstance(x, date) or isinstance(x, datetime):
+        return date_to_iso(x)
```
Comment (jstvz) [inline cumulusci/core/keychain/encrypted_file_project_keychain.py:57]: Is this what we want here?
```python
>>> date_to_iso(datetime.now())
'2022-10-13'
>>> datetime.now().isoformat() # 3.7+
'2022-10-13T13:58:56.902383'
>>> date.today().isoformat()  # 3.7+
'2022-10-13'
```

File: cumulusci/core/keychain/encrypted_file_project_keychain.py [hunk 2]
```diff
@@ -127,21 +130,23 @@ def _decrypt_config(self, config_class, encrypted_config, extra=None, context=No
...
             # Convert bytes created in Python 2
             config_dict = {}
-            for k, v in unpickled.items():
+            for k, v in config_obj.items():
                 if isinstance(k, bytes):
                     k = k.decode("utf-8")
                 if isinstance(v, bytes):
                     v = v.decode("utf-8")
+                if v and k == "date_created":
+                    v = iso_to_datetime(v)
```
Comment (jstvz) [inline cumulusci/core/keychain/encrypted_file_project_keychain.py:149]:
```python
>>> datetime.fromisoformat('2022-10-13T20:59:05.488737') # 3.7+
datetime.datetime(2022, 10, 13, 20, 59, 5, 488737)
```

File: cumulusci/tasks/bulkdata/dates.py [hunk 3]
```diff
@@ -84,6 +84,13 @@ def iso_to_date(s):
     return datetime.strptime(s, "%Y-%m-%d").date()
 
 
+def iso_to_datetime(s):
+    """Convert ISO8601 string to date object"""
+    if isinstance(s, datetime):
+        return s
+    return datetime.strptime(s, "%Y-%m-%d")
```
Comment (jstvz) [inline cumulusci/tasks/bulkdata/dates.py:91]:
```python
>>> iso_to_datetime('2022-10-13T20:59:05.488737')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in iso_to_datetime
  File "/Users/jestevez/.pyenv/versions/3.8.14/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/Users/jestevez/.pyenv/versions/3.8.14/lib/python3.8/_strptime.py", line 352, in _strptime
    raise ValueError("unconverted data remains: %s" %
ValueError: unconverted data remains: T20:59:05.488737
```

File: cumulusci/core/keychain/encrypted_file_project_keychain.py [hunk 4]
```diff
@@ -43,17 +44,28 @@
...
-
 scratch_org_class = os.environ.get("CUMULUSCI_SCRATCH_ORG_CLASS")
 if scratch_org_class:
     scratch_org_factory = import_global(scratch_org_class)  # pragma: no cover
 else:
     scratch_org_factory = ScratchOrgConfig
 
 
+def extract_dates(x):
+    if "$type" in x and x["$type"] == "date":
```
Comment (prescod) [inline cumulusci/core/keychain/encrypted_file_project_keychain.py:55]: This is fine but could have been written:

```
if x.get("$type") == "...":
```

File: cumulusci/core/keychain/encrypted_file_project_keychain.py [hunk 5]
```diff
@@ -43,17 +44,28 @@
...
+        return date.fromisoformat(x["$value"])
+    if "$type" in x and x["$type"] == "datetime":
+        return datetime.fromisoformat(x["$value"])
+    return x
```
Comment (prescod) [inline cumulusci/core/keychain/encrypted_file_project_keychain.py:59]:
```suggestion
    assert "$type" not in x, f"Unknown $type: {x['$type']}"
    return x
```

**Review by prescod — DISMISSED** [review]

**Review by jstvz — APPROVED** [review]

**Pull request merged**